components/batch-system/src/batch.rs

```diff
@@ -30,35 +30,61 @@ use crate::{
     config::Config,
     fsm::{Fsm, FsmScheduler, Priority},
     mailbox::BasicMailbox,
+    metrics::*,
     router::Router,
     scheduler::{ControlScheduler, NormalScheduler},
 };
 
 /// A unify type for FSMs so that they can be sent to channel easily.
 pub enum FsmTypes<N, C> {
-    Normal(Box<N>),
-    Control(Box<C>),
+    Normal((Box<N>, Instant)),
+    Control((Box<C>, Instant)),
     // Used as a signal that scheduler should be shutdown.
     Empty,
 }
-pub struct NormalFsm<N> {
+
+struct MetricsCollector<N: Fsm> {
+    timer: Instant, // time since polled
+    round: usize,   // how many round the fsm has been continuously polled
+    _phantom: std::marker::PhantomData<N>,
```

Comment:
```suggestion
    fsm_type: String,
```

How about adding a String? It makes the code cleaner.

Reply:
the static metric vector `get()` only accepts the enum instead of string